p2p/chain.py

```diff
@@ -421,9 +429,56 @@ def __init__(self,
             await self._handle_block_bodies(peer, list(cast(Tuple[eth.BlockBody], msg)))
         elif isinstance(cmd, eth.NewBlock):
             await self._handle_new_block(peer, cast(Dict[str, Any], msg))
+        elif isinstance(cmd, eth.GetBlockHeaders):
+            self._handle_get_block_headers(peer, cast(Dict[str, Any], msg))
+        elif isinstance(cmd, eth.GetBlockBodies):
+            self._handle_get_block_bodies(peer, cast(List[Hash32], msg))
+        elif isinstance(cmd, eth.GetReceipts):
+            self._handle_get_receipts(peer, cast(List[Hash32], msg))
+        elif isinstance(cmd, eth.GetNodeData):
+            self._handle_get_node_data(peer, cast(List[Hash32], msg))
         else:
             self.logger.debug("%s msg not handled yet, need to be implemented", cmd)
 
+    def _handle_get_block_headers(self, peer: ETHPeer, msg: Dict[str, Any]) -> None:
+        n = msg['block_number_or_hash']
+        if isinstance(n, bytes):
+            header = self.chaindb.get_block_header_by_hash(cast(Hash32, n))
+            n = header.block_number
```

Comment:
This pattern of setting `n` above and then overriding in here is hard for me to parse.  I initially thought there was a bug until I realized what was going on.

Can we do something like `raw_block_number = msg['block_number_or_hash']` and then have:

```python
if isinstance(raw_block_number, bytes):
    block_number = raw_block_number
elif isinstance(raw_block_number, int):
    ...
else:
    raise TypeError("...")
```

That way we explicitly handle both allowed types.  (Also gets rid of the single letter variable name).